services/motion/localizer.go

```diff
@@ -56,8 +61,62 @@ type movementSensorLocalizer struct {
 // NewMovementSensorLocalizer creates a Localizer from a MovementSensor.
 // An origin point must be specified and the localizer will return Poses relative to this point.
 // A calibration pose can also be specified, which will adjust the location after it is calculated relative to the origin.
-func NewMovementSensorLocalizer(ms movementsensor.MovementSensor, origin *geo.Point, calibration spatialmath.Pose) Localizer {
-	return &movementSensorLocalizer{MovementSensor: ms, origin: origin, calibration: calibration}
+func NewMovementSensorLocalizer(ms movementsensor.MovementSensor, origin *geo.Point, calibration spatialmath.Pose) (Localizer, error) {
+	if err := ValidateGeopoint(origin); err != nil {
+		return nil, err
+	}
+
+	if err := ValidatePose(calibration); err != nil {
+		return nil, err
+	}
+
+	return &movementSensorLocalizer{MovementSensor: ms, origin: origin, calibration: calibration}, nil
+}
+
+// ValidateGeopoint validates that a geopoint can be used for
+// motion planning.
+func ValidateGeopoint(gp *geo.Point) error {
+	if math.IsNaN(gp.Lat()) {
+		return errors.New("lat can't be NaN")
+	}
+
+	if math.IsNaN(gp.Lng()) {
+		return errors.New("lng can't be NaN")
+	}
+
+	return nil
+}
+
+// ValidatePose validates that a pose can be used for
+// motion planning.
+func ValidatePose(p spatialmath.Pose) error {
```

Comment:
turns out that if any of the X, Y, Z of the R3.Vector passed to create the pose are NaN X,Y,Z of the resulting pose will all be nil so in practice I've only been able to get this first error to trigger.